pallets/automation-price/src/lib.rs

```diff
@@ -1115,9 +1132,9 @@ pub mod pallet {
 				//
 				let price_matched_target_condtion =
 					if task.trigger_function == TRIGGER_FUNC_GT.to_vec() {
-						task.trigger_params[0] < this_task_asset_price.amount
+						task.trigger_params[0] < this_task_asset_price.value
 					} else {
-						task.trigger_params[0] > this_task_asset_price.amount
+						task.trigger_params[0] > this_task_asset_price.value
 					};
```

Comment:
I feel like for code re-use purpose there needs to be a wrapper function for the `is_price_condition_matched()`, since the price is compared earlier and here again for checking `PriceAlreadyMoved`. Moreover, when there are more than 2 enums in future, this `if else` condition needs to change.